src/compas_model/model/group_node.py

```diff
@@ -45,6 +47,207 @@ def __init__(self, name=None, geometry=None, attributes=None, parent=None):
             self._tree = parent._tree
 
         # --------------------------------------------------------------------------
-        # for debugging, the default name is the guid of th GroupNode
+        # for debugging, the default name is the guid of the GroupNode
         # --------------------------------------------------------------------------
-        self.name = name if name else str(self.guid)
+        self._name = None
+        self.name = name
+
+    # ==========================================================================
+    # Serialization
+    # ==========================================================================
+
+    @property
+    def data(self):
+        return {
+            "name": self.name,
+            "attributes": self.attributes,
+            "children": [child.data for child in self.children],
```

Comment:
seems wrong that no recursion is needed here, but recursion is needed in the from function